akka-typed/src/main/scala/akka/typed/cluster/sharding/ClusterSharding.scala

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.typed.cluster.sharding
+
+import akka.actor.ExtendedActorSystem
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ShardCoordinator.{ LeastShardAllocationStrategy, ShardAllocationStrategy }
+import akka.cluster.sharding.{ ClusterSharding => UntypedClusterSharding }
+import akka.cluster.sharding.{ ShardRegion => UntypedShardRegion }
+import akka.typed.internal.adapter.{ ActorRefAdapter, ActorSystemAdapter }
+import akka.typed.scaladsl.adapter.PropsAdapter
+import akka.typed.{ ActorRef, ActorSystem, Behavior, Extension, ExtensionId, Props }
+
+sealed case class ShardingEnvelope[A](entityId: String, message: A)
```

Comment:
why `sealed` and and not `final`?

Reply:
good point, will `final`

Reply:
Though perhaps in the future we may want to add features here... 
We could add things that would allow "dont deliver if took more than a minute" and weird stuff like that. Not sure if good idea, but sealed would be the way to go then

Reply:
then it shouldn't be a `case class`, and it could always start out as `final`